src/targets/gpu/include/migraphx/gpu/gemm_impl.hpp

```diff
@@ -24,28 +24,492 @@
+            solution_indices.resize(list_size);
+            auto arg_list_solutions = create_strided_batched_gemm_get_solutions_args(
+                ctx, input_args, solution_indices.data(), &list_size);
+            rocblas_invoke(&rocblas_gemm_strided_batched_ex_get_solutions, arg_list_solutions);
+        }
+        else
+        {
+            auto arg_list = create_gemm_ex_get_solutions_args(ctx, input_args, nullptr, &list_size);
+            rocblas_invoke(&rocblas_gemm_ex_get_solutions, arg_list);
+            solution_indices.resize(list_size);
+            auto arg_list_solutions = create_gemm_ex_get_solutions_args(
+                ctx, input_args, solution_indices.data(), &list_size);
+            rocblas_invoke(&rocblas_gemm_ex_get_solutions, arg_list_solutions);
+        }
+
+        double bestTime = std::numeric_limits<double>::max();
+        // Initialize to default solution index
+        rocblas_int bestSol = 0;
+        for(auto sol : solution_indices)
+        {
+            // Warmup: the first few calls to an op. may not be representative since there is
+            // more time taken initializing caches, etc. so we won't time them.
+            for(rocblas_int cc = 0; cc < cold_calls; ++cc)
+            {
+                run(ctx, input_args, sol);
+            }
+            double host_time = 0.0;
+            // Define the function to be timed
+            auto run_func    = [&]() {
+                run(ctx, input_args, sol);
+                ctx.finish();
+            };
+            for(rocblas_int hc = 0; hc < hot_calls; ++hc)
+            {
+                ctx.finish();
+                host_time += time<microseconds>(run_func);
+            }
+            // todo:  Measured time dropped from 20 us to about 6.7 us when I raised hot_calls from
+            // 1 to 11. The higher the hot_calls value, the faster per-call time up to at least 25,
+            // and increasing cold_calls makes little or no difference.  Why?
+            host_time /= hot_calls;
+            std::cout << "Solution index " << sol << ": " << host_time << " us" << std::endl;
+
+            // track current best
+            if(host_time < bestTime)
+            {
+                printf(" current best index  %d\n", sol);
+                bestSol  = sol;
+                bestTime = host_time;
+            }
+        }
+        std::cout << "Winner: " << bestSol << " in " << bestTime << " us" << std::endl;
+        return bestSol;
+    }
+
+    private:
+    size_t num_matrices;
+    rocblas_int m, n, k;
+    bool transa, transb;
+    T alpha, beta;
+    void* alpha_v = nullptr;
+    void* beta_v  = nullptr;
+    rocblas_gemm_flags int8_flag;
+    rocblas_int lda, ldb, ldc, ldd;
+    rocblas_int a_stride, b_stride, c_stride, d_stride;
+    rocblas_datatype compute_type, arg_type, output_type;
+    bool strided_batched = true, is_3inputs = true, compute_fp32 = true;
+    uint32_t flags = 0; //  optional gemm flags.
```

Comment:
is this used?

Reply:
It's been replaced with individual flag names for the various create...() calls.  I'll hold off on deleting it, though, until I'm sure I'm not going to refactor those yet again.